extensions/notebook/src/book/bookTocManager.ts

```diff
@@ -206,6 +207,11 @@ export class BookTocManager implements IBookTocManager {
 		}
 		if (isModified) {
 			await fs.writeFile(tocPath, yaml.safeDump(this.tableofContents, { lineWidth: Infinity, noRefs: true, skipInvalid: true }));
+			let toc: undoRedoToc = {
+				undo: this._initialToc.get(tocPath),
+				redo: this.tableofContents
+			};
+			this.tocFiles.set(tocPath, toc);
```

Comment:
I'm not sure about storing undo/redo entries like this - we should try to emulate the undo/redo service in core and have the service be the only thing that knows about undo/redo stuff. So it would have its own list of edit operations that is pushed to while the manager would just be in charge of the current state.